diff --git a/shim_et/BUCK b/shim_et/BUCK
index a1a9bdaf65..e7f2fa39a2 100644
--- a/shim_et/BUCK
+++ b/shim_et/BUCK
@@ -28,6 +28,15 @@ system_python_bootstrap_toolchain(
     visibility = ["PUBLIC"],
 )
 
+execution_platform(
+    name = "android-arm32",
+    cpu_configuration = "prelude//cpu:arm32",
+    os_configuration = "prelude//os:android",
+    # REVIEW: not sure if this is correct
+    use_windows_path_separators = host_info().os.is_windows,
+    visibility = ["PUBLIC"],
+)
+
 execution_platform(
     name = "android-arm64",
     cpu_configuration = "prelude//cpu:arm64",
diff --git a/tools/cmake/Utils.cmake b/tools/cmake/Utils.cmake
index 9fbab17728..6179fd3994 100644
--- a/tools/cmake/Utils.cmake
+++ b/tools/cmake/Utils.cmake
@@ -88,6 +88,8 @@ function(extract_sources sources_file)
     if(ANDROID_ABI)
       if("${ANDROID_ABI}" STREQUAL "arm64-v8a")
         set(target_platforms_arg "--target-platforms=shim_et//:android-arm64")
+      elseif("${ANDROID_ABI}" STREQUAL "armeabi-v7a")
+        set(target_platforms_arg "--target-platforms=shim_et//:android-arm32")
       elseif("${ANDROID_ABI}" STREQUAL "x86_64")
         set(target_platforms_arg "--target-platforms=shim_et//:android-x86_64")
       else()
diff --git a/backends/xnnpack/third-party/cpuinfo/CMakeLists.txt b/backends/xnnpack/third-party/cpuinfo/CMakeLists.txt
index aedc983..b211554 100644
--- a/backends/xnnpack/third-party/cpuinfo/CMakeLists.txt
+++ b/backends/xnnpack/third-party/cpuinfo/CMakeLists.txt
@@ -317,6 +317,8 @@ ELSE()
   TARGET_COMPILE_DEFINITIONS(cpuinfo INTERFACE CPUINFO_SUPPORTED_PLATFORM=0)
 ENDIF()
 
+target_compile_options(cpuinfo PRIVATE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0)
+
 ADD_LIBRARY(${PROJECT_NAME}::cpuinfo ALIAS cpuinfo)
 
 # support find_package(cpuinfo CONFIG)
