diff --git a/install_requirements.py b/install_requirements.py
index 0d44956c47..b115c2d0c6 100644
--- a/install_requirements.py
+++ b/install_requirements.py
@@ -100,7 +100,6 @@ def install_requirements(use_pytorch_nightly):
 
     LOCAL_REQUIREMENTS = [
         "third-party/ao",  # We need the latest kernels for fast iteration, so not relying on pypi.
-        "extension/llm/tokenizers",  # TODO(larryliu0820): Setup a pypi package for this.
     ]
 
     # Install packages directly from local copy instead of pypi.
diff --git a/requirements-dev.txt b/requirements-dev.txt
index 1743b142a4..2ba197b9c9 100644
--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -8,5 +8,4 @@ tomli  # Imported by extract_sources.py when using python < 3.11.
 wheel  # For building the pip package archive.
 zstd  # Imported by resolve_buck.py.
 certifi  # Imported by resolve_buck.py.
-lintrunner==0.12.7
-lintrunner-adapters==0.12.4
+
